src/main/lib/applications.mjs

```diff
@@ -1,26 +1,184 @@
+  // and generate an event to notify the UI
+
+  async #refreshApplications () {
+    const runningRuntimes = await this.#runtimeClient.getRuntimes()
+    await this.#createMissingApplications(runningRuntimes)
+    const apps = await this.#getApps()
+    const ret = []
+
+    for (const app of apps) {
+      const appForList = {
+        id: app.id,
+        name: app.name,
+        path: app.path,
+        running: false,
+        status: 'stopped',
+        platformaticVersion: app.lastPltVersion,
+        isLatestPltVersion: app.lastPltVersion === this.latestPlatformaticVersion,
+        runtime: null,
+        insideMeraki: false,
+        lastStarted: app.startedAt,
+        lastUpdated: app.updatedAt
+      }
+      const runtime = runningRuntimes.find((runtime) => runtime.projectDir === app.path)
+      if (runtime) {
+        appForList.running = true
+        appForList.status = 'running'
+        appForList.platformaticVersion = runtime.platformaticVersion
+        appForList.isLatestPltVersion = (runtime.platformaticVersion === this.latestPlatformaticVersion)
+        appForList.runtime = runtime
+        appForList.insideMeraki = !!this.#started[app.id]
+        // We need to update the lastPltVersion if unknown or it's different
+        if (!app.lastPltVersion || app.lastPltVersion !== runtime.platformaticVersion) {
+          await this.#mapper.entities.application.save({
+            fields: ['id', 'lastPltVersion'],
+            input: { id: app.id, lastPltVersion: runtime.platformaticVersion }
+          })
+        }
+      }
+      ret.push(appForList)
+    }
+
+    this.#applications = ret
+  }
+
+  #getRuntimeCliPath (appFolder) {
+    const pkgJsonPath = join(appFolder, 'package.json')
+    const _require = createRequire(pkgJsonPath)
+    return _require.resolve('@platformatic/runtime/runtime.mjs')
+  }
+
+  async startRuntime (id, env = {}) {
+    const app = await this.#mapper.entities.application.find({ where: { id: { eq: id } } })
+    if (!app || app.length === 0) {
+      throw new Error(`Application with id ${id} not found`)
+    }
+    const appFolder = app[0].path
+    await npmInstall(null, { cwd: appFolder }, logger)
+    const configFile = join(appFolder, 'platformatic.json')
+    const runtimeCliPath = this.#getRuntimeCliPath(appFolder)
+    const runtime = execa(
+      process.execPath, [runtimeCliPath, 'start', '-c', configFile],
+      { env, cleanup: true, cwd: appFolder }
+    )
+    runtime.stdout.pipe(process.stdout)
+    runtime.stderr.pipe(process.stderr)
+
+    const output = runtime.stdout.pipe(split(function (line) {
+      try {
+        const obj = JSON.parse(line)
+        return obj
+      } catch (err) {
+        console.log(line)
+      }
+    }))
+
+    const errorTimeout = setTimeout(() => {
+      throw new Error('Couldn\'t start server')
+    }, 30000)
+
+    for await (const messages of on(output, 'data')) {
```

Comment:
output is a stream. You can iterate directly because this would not manage backpressure